### PR TITLE
Issue(1147): Add option to pretty print html components

### LINKED ISSUE #1147: Pretty-print html() output by default
### What problem does this feature solve?
My team is using `@vue/test-utils` for snapshot testing components but the snapshots are hard to parse because the HTML string output from `.html()` is not formatted in a readable way. If `html()` pretty-printed the HTML it would be a lot easier to review changes to the component-generated HTML.

### What does the proposed API look like?
`html()` would pretty-print the returned HTML string by default, perhaps using something like https://github.com/prettydiff/prettydiff. If necessary, a user could bypass pretty-printing by passing an option to `.html()`, perhaps `.html(false)` or `.html({ prettyPrint: false })`.

### PR DESCRIPTION
**Pull request by addisonssense**
This PR is to implement pretty printing for html components. 

Fixes #1147

### CODE REVIEW

**Review by quinnlangille — APPROVED** [review]
Yes! I've been waiting for this 🥇

**Comment by eddyerburgh** [comment]
Thank you for the PR, as per my comment in the issue: can you make pretty-printing the default behavior?

**Comment by addisonssense** [comment]
@eddyerburgh done! I have updated the existing test to use the `prettyPrint: false` option. If its better to update the html being asserted instead of using the flag, I can definitely do that instead. I figured this method was less intrusive

**Review by bilyes — APPROVED** [review]
Very cool feature at a relatively low cost. Thanks for the initiative @addisonssense !

**Comment by briwa** [comment]
@addisonssense I think existing tests should be updated with `prettyPrint: true` instead of `false`, since it should reflect the default behavior of `.html`? It will be consistent for future tests that use `.html`, since they would probably be written with the default behavior in mind (pretty-printed).

Just my $0.02 ✌️ But anyhow, nice PR!

**Comment by addisonssense** [comment]
@briwa I think you make a good point. I'll do that 👍

**Comment by TheJaredWilcurt** [comment]
A few months ago I proposed a different approach to this. `jest-serializer-vue` has options built in, but the options do not format (or "pretty print") the code very well.

* https://github.com/eddyerburgh/jest-serializer-vue/issues/13

### TLDR:

* Change the default settings being passed in to `pretty` to these:
  ```js
  const options = {
    indent_char: ' ',
    indent_inner_html: true,
    indent_size: 2,
    inline: [],
    sep: '\n',
    unformatted: ['code', 'pre']
  };
  ```
* Allow users to pass in their own options object. (Not sure where they would do this, `package.json`?)
* Have instructions of how to pass in the object with the previous settings (so users can opt out of breaking changes).

This allows us to change the defaults again in the future without worrying about breaking people's tests, as the defaults can be documented for each version.

**Comment by eddyerburgh** [comment]
I intend to implement the changes in jest-serializer-vue but I haven't had time to work on it.

jest-serializer-vue will be formatted more heavily, whereas vue-test-utils can just use the defaults.

@addisonssense thanks for updating, but I wasn't clear in my previous comment. I don't think there should be an option to pretty-print or not. I don't think it should be configurable, and pretty printing should always be the default behavior. 

Can you update the PR to remove the `prettyPrint` option entirely?

**Comment by addisonssense** [comment]
@eddyerburgh done! Pretty printing is now the default. The option to disable it is removed.

**Comment by addisonssense** [comment]
@eddyerburgh it looks like my build is stuck on circleci. Would you mind restarting that step?

**Review by eddyerburgh — APPROVED** [review]
Thanks :)
